src/main/java/hidn/navada/product/ProductJpaRepo.java

```diff
@@ -4,7 +4,21 @@
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.util.List;
 
 public interface ProductJpaRepo extends JpaRepository<Product,Long> {
     Page<Product> findProductsByUser(User user, Pageable pageable);
+    Page<Product> findProductsByProductNameContains(String productName, Pageable pageable);
+
+    @Query("select p from Product p where p.productName like %:productName% and p.category.categoryId in (:categoryIds)")
+    Page<Product> searchProductsByNameAndCategory(@Param("productName") String productName, @Param("categoryIds") List<Long> categoryIds, Pageable pageable);
+
+    @Query("select p from Product p where p.productName like %:productName% and p.productCost between :lowerBound and :upperBound")
+    Page<Product> searchProductsByNameAndCost(@Param("productName") String productName, @Param("lowerBound") int lowerBound, @Param("upperBound") int upperBound, Pageable pageable);
+
+    @Query("select p from Product p where p.productName like %:productName% and p.category.categoryId in (:categoryIds) and p.productCost between :lowerBound and :upperBound")
+    Page<Product> searchProductsByNameAndCategoryAndCost(@Param("productName") String productName, @Param("categoryIds") List<Long> categoryIds, @Param("lowerBound") int lowerBound, @Param("upperBound") int upperBound,Pageable pageable);
```

Comment:
```suggestion
    
@Query(value = "select p from Product p where (:#{#options.productName} is null or p.productName like %:#{#options.productName}%) " +
            "and (:#{#options.categoryId} is null or p.category.id in :#{#options.categoryIds}) " +
            "and (:#{#options.lowerBound} is null or (:#{#options.lowerBound} <= p. productCost)) " +
            "and (:#{#options. upperBound} is null or (p. productCost <= :#{#options.upperBound})) ")
    Page<Product> findProductsByOptions(@Param("options") ProductSearchOptions options, Pageable pageable);
```

이런식으루...?? 저두 어떤 방식이 나을지 몰라서 일단 생각나서 혹시 도움될까 싶어 일단 말해봤어욤...!! 핳핳